nifi-nar-bundles/nifi-marklogic-bundle/nifi-marklogic-processors/src/test/java/org/apache/nifi/marklogic/processor/PutMarkLogicTest.java

```diff
@@ -197,6 +197,56 @@ public void customPermissions() {
         assertEquals(DocumentMetadataHandle.Capability.UPDATE, perms.get("manage-admin").iterator().next());
     }
 
+    @Test
+    public void twoPermissionsWithSameRole() {
+        processContext.setProperty(PutMarkLogic.PERMISSIONS, "manage-user,read,manage-user,update");
+        processor.initialize(initializationContext);
+
+        addFlowFile("<test/>");
+
+        processor.onTrigger(processContext, mockProcessSessionFactory);
+
+        DocumentMetadataHandle metadata = (DocumentMetadataHandle) processor.writeEvent.getMetadata();
+        DocumentMetadataHandle.DocumentPermissions perms = metadata.getPermissions();
+        assertEquals(1, perms.size());
+        Iterator<DocumentMetadataHandle.Capability> iterator = perms.get("manage-user").iterator();
+        assertEquals(DocumentMetadataHandle.Capability.READ, iterator.next());
```

Comment:
When I copied these changes into my fork and built with the 1.8.0.x branch, this test failed because the capabilities came back in a different order. Consider replacing lines 213-214 with:

```java
        java.util.Set<DocumentMetadataHandle.Capability> capabilities = perms.get("manage-user");
        assertEquals(2, capabilities.size());
        assertTrue(capabilities.contains(DocumentMetadataHandle.Capability.UPDATE));
        assertTrue(capabilities.contains(DocumentMetadataHandle.Capability.READ));
```

Reply:
Nice catch - I made the updates, merging this into develop.